resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsServerAddressStreamProvider.java

```diff
@@ -117,10 +119,21 @@
                         "Default DNS servers: {} (sun.net.dns.ResolverConfiguration)", defaultNameServers);
             }
         } else {
-            Collections.addAll(
-                    defaultNameServers,
-                    SocketUtils.socketAddress("8.8.8.8", DNS_PORT),
-                    SocketUtils.socketAddress("8.8.4.4", DNS_PORT));
+            // Depending if IPv6 or IPv4 is used choose the correct DNS servers provided by google:
+            // https://developers.google.com/speed/public-dns/docs/using
+            // https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html
+            if (NetUtil.isIpV6AddressesPreferred() ||
+                    (NetUtil.LOCALHOST instanceof Inet6Address && !NetUtil.isIpV4StackPreferred())) {
+                Collections.addAll(
+                        defaultNameServers,
+                        SocketUtils.socketAddress("2001:4860:4860::8888", DNS_PORT),
+                        SocketUtils.socketAddress("2001:4860:4860::8844", DNS_PORT));
+            } else {
+                Collections.addAll(
+                        defaultNameServers,
+                        SocketUtils.socketAddress("8.8.8.8", DNS_PORT),
+                        SocketUtils.socketAddress("8.8.4.4", DNS_PORT));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this IP "8.8.4.4" address configurable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1313)